### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by wilsjame**
Changed all instances of "a" before a vowel sound to "an". For example, changing "a 8-bit" to "an 8-bit".

### CODE REVIEW

File: src/CGB_Registers.md
Comment (ISSOtm) [inline src/CGB_Registers.md:93]: It's arguably a weird/confusing/tricky rule, but "a" should be used before a sounded "H", not "an".

File: src/CGB_Registers.md
Comment (ISSOtm) [inline src/CGB_Registers.md:107]: Same here.

File: src/OAM.md
Comment (ISSOtm) [inline src/OAM.md:18]: While we're at this,
```suggestion
Y=2 hides an 8×8 sprite but displays the last two rows of an 8×16 sprite,
```

File: src/OAM.md
Comment (ISSOtm) [inline src/OAM.md:20]:
```suggestion
Y=144 displays an 8×16 sprite aligned with the bottom of the screen,
```

File: src/OAM.md
Comment (ISSOtm) [inline src/OAM.md:21]:
```suggestion
Y=152 displays an 8×8 sprite aligned with the bottom of the screen,
```

File: src/Power_Up_Sequence.md
Comment (ISSOtm) [inline src/Power_Up_Sequence.md:34]: `a` here is a  register, so it shouldn't be changed.

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Thanks for the cleanup! Appreciate this, as it helps making the document feel more professional and pleasant to read ^^

File: src/SGB_Command_Undocumented.md
Comment (avivace) [inline src/SGB_Command_Undocumented.md:3]: I don't think SGB is pronounced as an acronym here, but as Super Game Boy

File: src/SGB_Functions.md
Comment (avivace) [inline src/SGB_Functions.md:6]: Again, I don't think there's a clear consesus on how to pronounce SNES so I wouldn't go as far as changing these occurences

**Review by avivace — CHANGES_REQUESTED** [review]
SGB and SNES are words pronounced in both ways (as acronyms, hence having a vowel sound and requiring the `an` and expanding the words, so having the S sound needing the `a`) so I wouldn't change them.

Everything else looks like a nice improvement.

File: src/SGB_Functions.md
Comment (pinobatch) [reply src/SGB_Functions.md:6]: *Nintendo Power* magazine abbreviated it as "Super NES" most often. I even vaguely remember a letter to the editor in Player's Pulse asking about this, with a reply from the editor to the effect "our style guide says 'Super NES'."

File: src/CGB_Registers.md
Comment (pinobatch) [reply src/CGB_Registers.md:93]: User voxel#9080 in the gbdev server on Discord raised a question about "an aitch blank" (in the lect of the child actress in the ["A, B, C, D, PCP" anti-drug PSA](https://www.youtube.com/watch?v=Wob_38j5JV8)) vs. "a haitch blank" (in voxel's lect). Surprisingly, ["Haitch or aitch? How a humble letter was held hostage by historical haughtiness" by Kate Burridge and Catherine McBride](https://theconversation.com/haitch-or-aitch-how-a-humble-letter-was-held-hostage-by-historical-haughtiness-97184) argues for a shift toward "haitch" on grounds that children learn letters whose name starts with the sound that it makes more quickly.

Comment (aaaaaa123456789) [reply src/CGB_Registers.md:93]: Because of the pronunciation differences, both "a" and "an" would be correct in this context, of course.

File: src/Power_Up_Sequence.md
Comment (aaaaaa123456789) [reply src/Power_Up_Sequence.md:34]: Something tells me this was the result of a regex...

File: src/CGB_Registers.md
Comment (wilsjame) [reply src/CGB_Registers.md]: I changed it back to "a" to avoid unnecessary commits when both usages are arguably correct. The article was really interesting!

Comment (wilsjame) [reply src/CGB_Registers.md]: Thank you for the feedback ^_^ marked as resolved.

File: src/Power_Up_Sequence.md
Comment (wilsjame) [reply src/Power_Up_Sequence.md]: Ah I see, nice! Haha yeah it was, a combination of grep and brute force searching in vim 😅. I pushed a fix.

File: src/SGB_Functions.md
Comment (wilsjame) [reply src/SGB_Functions.md]: Thanks for the feedback, good points. After searching around, it appears "a SNES" is more used than "an SNES". On the SNES Wikipedia page, "a SNES-CD" is used. I pushed a commit undoing the change. 

Leaving this conversation unresolved in case @pinobatch's point about using "Super NES" wants to be addressed further. Otherwise, a maintainer can mark this as resolved.

File: src/SGB_Command_Undocumented.md
Comment (wilsjame) [reply src/SGB_Command_Undocumented.md]: That's right, thanks for the review. I pushed a commit undoing the change.

**Review by avivace — APPROVED** [review]
Looks okay! Thanks.

Just a little side note for the future: please refrain from using regex or mindless grepping for these cases without reviewing them. They can lead to (as you've seen in those discussions) false positives and gray situations where it's better to just leave the general "neutral" rule in place instead of implying pronunciations or habits of a quite wide audience :D

**Comment by wilsjame** [comment]
Thank you for the help! I see now how commiting every instance causes extra time for everyone to get things in order. Commiting only the clear instances and asking about the neutral ones is a better approach. Appreciate everyone bearing with me and the feedback ^_^ these docs are an awesome resource!

**Review by ISSOtm — APPROVED** [review]
Yup, I'm good with all that's left. Running CI, will merge if that and Pino's review are green.

**Pull request merged**
